Step3 리뷰요청드립니다

zerodkdj opened the pull request.

zerodkdj committed:
docs: 로또기능 README.md파일에 정리

zerodkdj committed:
test: 로또클래스 테스트코드작성

- 로또가격조회,로또번호조회
- 번호개수 6개아니면 예외발생
- 중복번호있으면 예외발생
- 번호가 1에서 45사이가 아니면 예외발생

zerodkdj committed:
test: 로또컬렉션 테스트코드작성

- 자동으로 추가했을때 번호조회할 수 있는지 테스트
- 구매한 개수만큼 가격이 조회되는지 테스트

zerodkdj committed:
docs: README.md파일에 로또컬렉션 기능추가

zerodkdj committed:
test: 로또를 가격만큼 구매한뒤에 로또컬렉션에서 조회가능한지 테스트

- LottoService클래스 생성후 구매메소드추가

zerodkdj committed:
test: 당첨번호를 입력하면 로또번호가 몇등인지조회

- Rank상수를 enum타입으로 생성해서 상태값과 필요한로직 캡슐화

zerodkdj committed:
refactor: 로또컬렉션클래스 수정

- getSumOfPriceLotto명확성을 위해 getSumOfPriceLottos로 끝에 s추가
- 테스트를 위해 Lottos에서 생성자로 List<Lotto>를 입력받을 수 있게 변경

zerodkdj committed:
refactor: 로또컬렉션클래스에서 구매한 로또 번호조회하는 메소드이름 getNumbersOfLottos로 수정

zerodkdj committed:
test: 당첨번호입력했을때 당첨통계조회하는 getRankStatsOfLottos메소드 테스트

zerodkdj committed:
fix: 3등 당첨된경우 통계조회하는 테스트코드 검증로직수정

zerodkdj committed:
docs: README.md파일에 로또컬렉션,로또서비스 기능추가

zerodkdj committed:
refactor: mvc기반으로 패키지구분

zerodkdj committed:
feat: 로또진행을 위한 Controller,화면출력을 위한 View클래스 생성

- 로또 진행순서에 맞게 Controller로직추가
- 입력받기위한 View클래스 메소드 추가

zerodkdj committed:
feat: 화면입력을 위한 Console클래스추가

- int,String입력을 받을 메소드추가

zerodkdj committed:
feat: 결과출력을위한 View,Dto클래스 추가

- ResultView에서 구매내역,당첨통계출력
- ResultView에서 사용하기위한 관련Dto클래스추가

zerodkdj committed:
test: getRankStatus메소드에서 당첨결과 테스트

zerodkdj committed:
refactor: LottoController수정

- 입력받는부분 Console클래스로처리

zerodkdj committed:
style: 코드에 공백정리

zerodkdj committed:
refactor: LottoService클래스 생상자에서 Lottos파라미터 삭제

- 생성자 내부에서 팩토리메서드로 객체생성
- 가격대비 개수만큼상 생성하기위해 LottoFactory클래스에서 로또생성 담당

zerodkdj committed:
refactor: Lottos클래스에서 List에 Lotto객체추가하는 addByAuto메서드삭제

- Lotto객체추가는 생성자에서 이미 생성
- LottoService클래스에서 로또생성하는 메소드삭제
- Lottos에서하고있는 로또생성로직 삭제

zerodkdj committed:
test: LottoServiceTest 각 당첨별로 상금비교 테스트

zerodkdj committed:
style: 코드스타일통일

zerodkdj committed:
style: 코드스타일통일

zerodkdj committed:
refactor: LottoNumber클래스추가

- 기존에 Integer로 받아서 처리하던 로또번호를 LottoNumber타입으로 수정

zerodkdj committed:
refactor: LottoNumber클래스추가

- 기존에 Integer로 받아서 처리하던 로또번호를 LottoNumber타입으로 수정

zerodkdj committed:
refactor: Lotto클래스 getRank메서드 디버그용코드 제거

zerodkdj committed:
refactor: Lotto클래스 getRankStatsOfLottos메소드 안쓰는 코드정리

zerodkdj committed:
refactor: Rank클래스 조회메소드이름 컨벤션따라서 valueOf로 변경

zerodkdj committed:
refactor: LottoStatusDto클래스이름 명확하게하기위해 LottoResultDto로변경

zerodkdj committed:
refactor: LottoStatusDto클래스이름 명확하게하기위해 LottoResultDto로변경

zerodkdj committed:
fix: Lottos,LottoService에서 수정된 메소드, 테스트코드에도 반영

zerodkdj committed:
test: Rank클래스 테스트추가

- valueOf값에 대한 반환값 확인

zerodkdj committed:
style: 소스코드에 메소드간 공백추가

seokhongkim commented on src/main/java/step3/controller/LottoController.java:
```diff
@@ -0,0 +1,34 @@
+
+
+    public void start() {
+        InputView.printPurchasingAmount();
```
1. `InputView` 와 `Console` 의 차이점은 무엇일까요?
2. 컨트롤러 입장에서는 `금액을 받아오는 것` 이 중요하지, 콘솔에 무슨 내용을 출력하는지는 관심사가 아닐 것 같습니다. 현재 콘솔 기반의 프로그램을 웹 환경으로 제공한다고 생각해보면, 컨트롤러는 금액만 잘 받아오면 되지 않을까요?

zerodkdj replied:
프리코스때부터 고민한 부분입니다. 써주신 글를 제가 이해한게 맞다면
금액을 받아오는 부분에 출력하는 코드를 같이 작성해주는게 맞을까요??
말씀하신 부분을 LottoConsoleView라는 클래스로 만들어봤습니다. 리뷰한번 더 부탁드립니다:)

seokhongkim replied:
네, 저는 지금 수정하신 코드처럼 `LottoConsoleView` 에서 입력을 요청하는 문구 출력도 같이 작성해주는 게 좋다고 생각합니다.
어떤 방법으로 사용자에게 입력을 요청하고 값을 읽어오는지는 `LottoController` 가 몰라도 되는 세부사항이라고 생각합니다. 또한, 데이터 입력을 요청하는 문구 `출력`과 데이터 `입력`을 받는 기능 모두 `콘솔`에 의존하고 있으므로, 하나의 클래스에서 구현해도 괜찮다고 생각합니다.
반면, 결과를 출력하는 것은 `데이터를 출력하기 위한 가공`의 책임을 가지고 있으므로, 입력을 요청하는 클래스와 별도의 클래스에서 관리해도 괜찮다고 생각합니다. :)

seokhongkim commented on src/main/java/step3/controller/LottoController.java:
```diff
@@ -0,0 +1,34 @@
+    public void start() {
+        InputView.printPurchasingAmount();
+        int amount = Console.readInt();
+        LottoService lottoService = new LottoService(amount);
```
하나의 메소드 내에 있는 코드도 빈 줄을 이용하여 적절한 문단 단위로 구분하면, 코드를 읽고 이해하기가 더 쉬울 것 같아요.
예를 들면, 컨트롤러의 로직은 아래와 같아 보입니다.

```
금액을 입력받는다.

로또를 자동으로 구매한다.
구매한 로또를 출력한다.

우승 번호를 입력받는다.
로또 결과를 계산한다.
로또 결과를 출력한다.
```

seokhongkim commented on src/main/java/step3/io/Console.java:
```diff
@@ -0,0 +1,27 @@
+package step3.io;
```
`io` package 와 `view` package 의 차이는 무엇인가요? :)

seokhongkim commented on src/main/java/step3/model/LottoNumber.java:
```diff
@@ -0,0 +1,61 @@
+    static {
+        LOTTO_NUMBERS = IntStream
+                .rangeClosed(LOTTO_MIN_VALUE, LOTTO_MAX_VALUE)
+                .boxed()
```
wrapper class 로 변경하신 이유가 있을까요?

seokhongkim commented on src/main/java/step3/model/LottoNumber.java:
```diff
@@ -0,0 +1,61 @@
+
+    private static final String OUT_OF_NUMBER_MESSAGE = "번호는 1에서 45사이만 허용합니다";
+
+    private static List<LottoNumber> LOTTO_NUMBERS;
```
미리 캐시로 값을 만들어놓고 사용하시는 것 좋습니다. 👍 
캐시 생성 후에는 데이터 변경이 불가하도록 `ImmutableList` 로 만들어서 관리해보셔도 좋겠습니다. :)

seokhongkim commented on src/main/java/step3/model/LottoNumber.java:
```diff
@@ -0,0 +1,61 @@
+public class LottoNumber {
+
+
+    public static final int LOTTO_MIN_VALUE = 1;
```
로또 번호의 최솟값과 최댓값을 외부에서 접근할 수 있도록 공개해놓으신 이유가 있을까요?

zerodkdj replied:
이 부분은 LottoFactory에서 참조하기 위해서 public으로 작성한건데 리팩토링하면서 private으로 고쳤습니다!

seokhongkim commented on src/main/java/step3/model/LottoFactory.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    public static List<Lotto> createLottos(int size) {
+        if (size <= 0) throw new IllegalArgumentException(NONE_POSITIVE_MESSAGE);
```
꼼꼼한 밸리데이션 좋습니다. 👍 
다만, `LottoFactory` 는 `Lotto` 도메인을 알고 있고, `Lotto` 도메인의 규칙은 6개의 숫자로 이루어져야 한다는 것이니 개수가 6개가 아니면 문제가 되는 것은 아닐까요?
0보다 작거나 같은 경우 문제가 되는 것으로 구현하신 이유가 궁금합니다.

zerodkdj replied:
6개가 아니면 문제가 발생하는건 Lotto도메인에서 생성할때 검증중입니다. 물어보신 '0보다 작거나 같은 경우 문제가 되는 것' 은 사실 밑에 리뷰에써주신 '1장도 구매하지 않는 경우에 대한 검증'입니다.

궁금한점이 한장도 구매하지 않는 경우에 대한 검증을 size값을 생성한 LottoMachine에서 하는게 좋을까요 아니면 지금 코드에 있는것처럼 size값을 사용하는  createLottos에서하는게 좋을까요??? 

리뷰를 보니까 생성한곳에서 하는게 좋은거같기도 한데 헷갈리네요..일단 생성한곳에서 검증코드 넣는걸로 수정했습니다 :)

seokhongkim replied:
1장도 구매하지 않는 경우에 대한 검증은 `LottoMachine`, `LottoFactory` 중 어디에 있어도 자연스러운 것 같네요!
`LottoMachine` 의 입장에서는 입력된 금액이 없으니 비정상적인 상황으로 생각되고, `LottoFactory` 의 입장에서는 0장을 만들어라달라는 요청이 비정상적인 상황으로 보입니다.

seokhongkim commented on src/main/java/step3/model/LottoFactory.java:
```diff
@@ -0,0 +1,42 @@
+    private static final String NONE_POSITIVE_MESSAGE = "로또 개수는 한개 이상만 허용됩니다";
+
+    static {
+        CANDIDATE_NUMBERS = IntStream
```
혹시 `LottoFactory` 에서 발급할 수 있는 번호 목록을 따로 관리하시는 이유가 있을까요?
생성될 수 있는 값이 `LottoNumber` 에 이미 `LOTTO_NUMBERS` 로 만들어져서 관리되고 있으니, `LottoNumber` 에게 이 값을 요청해서 `shuffle` 하는 방법을 사용해보면 어떨까요?

zerodkdj replied:
실수한거 같습니다. LottoFactory에 번호목록이 LottoNumber에 중복되있었습니다ㅜㅜ  써주신대로 수정했습니다 :)

seokhongkim commented on src/main/java/step3/model/Lotto.java:
```diff
@@ -0,0 +1,44 @@
+
+public class Lotto {
+
+    public static final int PRICE = 1000;
```
여기도 금액과 번호 개수의 접근 제어 레벨을 public 으로 설정하신 이유가 있을까요?

zerodkdj replied:
LottoMachine에서 가격을 입력받으면 몇개를 만들지 계산하기위해 로또개당 가격을 알아야하는데 수정할 수 없는 final타입이기떄문에 외부에서 접근할 수 있게 public으로 만들었습니다. getMethod를통해 제공하도록 수정했는데 어떻게 생각하시나요?? :)

seokhongkim replied:
`PRICE` 가 사용되는 곳을 봤을 때 `LottoMachine` 에서 가장 많이 사용하고, `Lottos` 에서도 사용되는 것으로 보이네요.
`Lottos` 는 `LottoMachine` 에서 생성해주므로 값을 전달해줄 수 도 있을 것 같습니다. 그렇다면 현재 구조에서는 `LottoMachine` 이 금액을 가장 잘 알아야 하는 책임이 있을 것으로 보이기도 하네요.

또는 총 구매한 금액을 `LottoMachine` 에서 별도의 객체로 관리하고 있어도 좋을 것 같아요. 해당 객체는 입력한 금액에 대한 검증을 관리해볼 수 있겠네요! 금액에 대한 책임을 가지고 있으니, 로또 한장의 가격도 이 객체가 관리해도 좋을 것 같습니다. :)

zerodkdj replied:
돈을 관리하는 LottoMoney객체를 별도로 만들어서 관리하니까 코드가 훨씬 깔끔해졌습니다!!  감사합니다 :)

seokhongkim commented on src/main/java/step3/model/LottoService.java:
```diff
@@ -0,0 +1,38 @@
+
+    public LottoService(int purchasePrice) {
+        this.purchasePrice = purchasePrice;
+        int size = purchasePrice / Lotto.PRICE;
```
금액이 1000원 단위가 아닌 경우에 대한 검증, 1장도 구매하지 않는 경우에 대한 검증도 해보시면 좋겠습니다.

zerodkdj replied:
가격검증메소드에서 가격이 1000원단위로 입력됬는지 검증과 가격이 1000원 미만인지를 검증하면 size에 대한 검증도 같이 할 수 있어 메소드를 하나로 만들었는데 어떻게 생각하시나요 ?? :)

seokhongkim commented on src/main/java/step3/model/LottoService.java:
```diff
@@ -0,0 +1,38 @@
+
+public class LottoService {
+
+    private final int purchasePrice;
```
`LottoService` 의 역할은 무엇인가요?
`purchasePrice`, `lottos` 를 필드로 들고 있는데 `Service` 라는 이름을 하고 있으면 도메인의 규칙을 관리하기 위한 클래스가 아닌 절차를 서술하기 위한 클래스로 보입니다.

seokhongkim commented on src/main/java/step3/model/dto/LottoResultDto.java:
```diff
@@ -0,0 +1,29 @@
+    public List<RankDto> getRanks() {
+        return ranks.stream()
+                .sorted(Comparator.comparingInt(RankDto::getMatchCount))
+                .filter(rankDto -> rankDto.getMatchCount() > 0)
```
`MatchCount > 0` 인 경우를 나타낼 수 있는 도메인 용어는 무엇일까요?
이 로직은 `당첨된 등수별 개수` 를 계산하기 위한 로직으로 보이는데요. 맞다면, 이를 설명하기 위한 이름을 사용해보면 어떨까요? :)

seokhongkim commented on src/main/java/step3/model/dto/LottoResultDto.java:
```diff
@@ -0,0 +1,29 @@
+                .collect(Collectors.toList());
+    }
+
+    public double getWinnigPercent() {
```
수익률 계산도 로또 도메인이 알아야 하는 내용은 아닐까요?

seokhongkim commented on src/main/java/step3/view/ResultView.java:
```diff
@@ -0,0 +1,30 @@
+        ranks.stream().forEach(rankDto -> {
+            System.out.printf("%d개 일치 (%d원)- %d개\n", rankDto.getMatchCount(), rankDto.getWinningPrice(), rankDto.getWinningCount());
+        });
+        System.out.printf("총 수익률은 %.2f입니다", (Math.floor(statusDto.getWinnigPercent() * 100) / 100.0));
```
수익률이 마이너스일 때 출력되는 `(기준이 1이기 때문에 결과적으로 손해라는 의미임)` 문구가 없는 것 같아요!

seokhongkim commented on src/test/java/step3/LottoServiceTest.java:
```diff
@@ -0,0 +1,198 @@
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mockStatic;
+
+public class LottoServiceTest {
```
꼼꼼한 단위 테스트 좋습니다. 👍 
mockito 를 사용하지 않고 테스트를 작성해보시면 좋겠습니다. :)

seokhongkim reviewed (changes requested):
동기님, 3단계 미션 잘 구현해주셨습니다. 👍 

조금 더 고민해보시면 좋을 것 같은 내용이 있어 코멘트를 남겼습니다.
코멘트 확인 부탁드립니다. 🙏 

코멘트 보시고 궁금하신 점이나 함께 이야기해보면 좋을 내용이 있다면 깃헙 코멘트 또는 슬랙 DM으로 편하게 말씀 부탁드립니다. 😀

zerodkdj committed:
test: RankDto클래스 isWin메소드에서 맞춘번호가 3개 이상일경우 true반환 테스트

zerodkdj committed:
refactor: LottoService클래스이름 역할의 명확성을위해 LottoMachine클래스로 수정

zerodkdj committed:
refactor: 수익률계산기능 LottoMachine으로이동

- Dto는 도메인에서 처리된 로직결과만 보여주는 기능

zerodkdj committed:
refactor: LottoResultDto클래스 수익률조회메소드 변수명에 맞게 getWinnigPercent에서 getPriceRatio로 수정

- 관련 테스트코드도 같이 수정

zerodkdj committed:
refactor: LottoNumber클래스 번호생성로직수정

- 스트림에서 boxed() 사용하지않고 mapToObject로 한번에 로또번호 생성

zerodkdj committed:
refactor: Console,InputView,ResultView삭제후 LottoConsoleView로 통합

zerodkdj committed:
style: 코드스타일통일

- 람다식 메소드참조로바꾸기
- 코드문맥간(변수,생성자,메소드) 공백으로 구분

zerodkdj committed:
refactor: 입력값 로또번호로 만들어주는 파싱클래스 패키지명 io에서 parser로 수정

seokhongkim commented on src/main/java/step3/controller/LottoController.java:
```diff
@@ -0,0 +1,32 @@
+
+
+    public void start() {
+        int amount = LottoConsoleView.printPurchasingAmount();
```
현재 메소드 이름은 `출력`을 주요 동작으로 서술하는 것으로 보이는데, 어떻게 생각하시나요? :)
저는 `구매 금액을 입력` 받고, `우승번호를 입력` 받는 기능이 각 메소드의 핵심 동작일 수 있겠다는 생각이 들었습니다.

zerodkdj replied:
저도 메소드이름 시작이 print라고 시작되어 출력만 담당하는것처럼 보인다고 생각되네요
출력담 담당하는 메소드를 LottoOutputView라고 따로 클래스를 만들어서 출력하는 메소드이름는 print~, 입력을 받는 메소드이름은 input~ 이런식으로 고쳐봤는데 어떻게 생각하시나요??? :)

seokhongkim replied:
입력이 주요 역할인 클래스와 출력이 주요 역할인 클래스를 분리하신 것 좋습니다. 👍 
메소드 이름도 무슨 기능을 하는지 더 잘 이해되네요!

seokhongkim commented on src/main/java/step3/controller/LottoController.java:
```diff
@@ -0,0 +1,32 @@
+public class LottoController {
+
+
+    public void start() {
```
빈 줄이 추가되어 코드의 가독성이 훨씬 좋아진 것 같아요! 👍

zerodkdj replied:
훨씬 깔끔해진거 같습니다!! 감사합니다 :)

seokhongkim commented on src/main/java/step3/view/LottoConsoleView.java:
```diff
@@ -0,0 +1,49 @@
+        return SCANNER.next();
+    }
+
+    public static void printPurchasingLottos(LottosNumberDto lottosNumberDto) {
```
로또 번호를 정렬해서 출력해줘야 할 것 같은데, 정렬에 대한 로직은 어디 있을까요? :)

zerodkdj replied:
LottoResultDto클래스의 getRanks메소드에 구현했습니다. 처음에는 Lottos나 Lotto,Rank같은 도메인에서 comparable을 구현할 생각이었는데 출력에 관한 요구사항이라서(화면에따라 내림차순으로 보여주는 요구사항이 있을수도있다고 생각)출력데이터를 가지고 있는 LottoResultDto클래스의 getRanks메소드에서 구현했습니다. 

그런데 출력데이터기때문에 출력을 담당하는 LottoConsoleView에서 정렬을 해야될까 고민이 되는데 어떻게 생각하시나요?? 저는 View에서 로직을 최대한 빼고싶어서 Dto에서 구현하긴했습니다.

seokhongkim replied:
이 코멘트는 발급한 로또 번호를 출력하는 것에 대한 내용입니다. :)
말씀하신 등수 출력에 대해서는 정렬이 잘 되는 것을 확인했습니다!

출력을 `LottoConsoleView` 와 `Dto` 중 어디서 하는 것이 좋은지에 대해서, 저는 지금은 두 곳 모두 괜찮다고 생각합니다.
하지만 규모가 커질수록 `LottoConsoleView` 로 옮기는 것이 좋을 것 같다는 생각입니다. 이유는 아래와 같습니다.
1. `Dto` 는 데이터 전달을 위한 목적이므로 로직을 적게 가지고 있는 것이 좋다고 생각하고, `LottoConsoleView` 는 화면 출력에 대한 책임을 가지고 있으므로 어떤 형태로 보여줄것인지에 대한 책임도 가지고 있다고 생각합니다. 지금은 `Dto` 가 `LottoConsoleView` 만을 위해 사용되므로, 강하게 결합되어 있어도 큰 영향은 없다고 생각합니다.
2. 콘솔 프로그램이지만 웹 환경으로 확장을 생각해봤을 때, 출력과 관련된 책임은 프론트엔드에서 가지고 있습니다. 성능상의 이유나 다른 이유로 백엔드에서 프론트엔드가 원하는 형태로 데이터를 가공하는 과정이 일부 있지만, 실제로 데이터가 잘 보여져야 하는 것은 프론트엔드의 관심사와 책임이라고 생각합니다. 하나의 `Dto` 가 여러 화면에서 사용되거나, 혹은 다양한 `Dto` 를 하나의 화면에서 보여줘야 하는 상황이 발생할 수 있는데, 이때 화면이 출력에 대한 책임을 가지고 있다면 확장과 재사용이 더 좋을 것 같습니다.

zerodkdj replied:
생각해보니까 View클래스에 책임이 어떻게 보여주는지까지 포함되는거같습니다. 감사합니다!! :)

seokhongkim commented on src/main/java/step3/model/dto/LottosNumberDto.java:
```diff
@@ -0,0 +1,23 @@
+
+    private final List<List<Integer>> lottosNumber;
+
+    public LottosNumberDto(List<List<LottoNumber>> lottosNumber) {
```
`Lotto` 가 `List<LottoNumber>` 를 가지고 있고, `Lottos` 는 `List<Lotto>` 를 가지고 있어서 이 DTO와 동일한 데이터를 가지고 있을 것으로 보이는데요.
`List<List<LottoNumber>>` 로 데이터를 관리하신 이유가 있을까요?

zerodkdj replied:
생각하지 못한 부분입니다 Lottos로 받아서 수정해봤습니다! Lottos로 받으면 Dto에 요구사항변화가 있을때 수정범위가 Dto로 줄어들어서 더 괜찮은거같습니다 :)

seokhongkim reviewed (changes requested):
동기님 안녕하세요! 3단계 미션 피드백 잘 반영해주셨습니다. 💯 

일부 피드백에 대해서는 별도로 말씀하시거나 작업하신 내용이 없는 것 같아서, 혹시 아래 접힌 부분에 있는 코멘트도 확인하신 것인지 여쭤봅니다.
<img width="837" alt="스크린샷 2022-10-29 오후 3 13 39" src="https://user-images.githubusercontent.com/44194568/198817115-eaa4e39a-f8e1-4eee-baa9-3188756072a3.png">

이외에 수정하신 코드에 코멘트를 조금 더 남겼습니다.
코멘트 확인 부탁드립니다. 🙏

zerodkdj committed:
refactor: LottoFactory의 중복된 static변수삭제

- LottoFactory의 로또의 번호값을 나타내는 CANDIDATE_NUMBERS는 LottoNumber클래스에서 알 수 있기때문에 중복코드문제로 삭제

zerodkdj committed:
refactor: Lotto클래스의 public변수 private으로수정

- Lotto클래스에서 public변수를 사용하고있는 나머지 클래스들은 get메소드를 통해 접근

zerodkdj committed:
refactor: LottoFactory클래스의 size값 검증하는로직 LottoMachine으로 이동

- size값을 생성한곳에서 검증

zerodkdj committed:
refactor: LottosNumberDto생성자 파라미터 Lottos로 수정

- 요구사항 변화가 있을때 수정범위가 줄어들 수 있게 Lottos로 수정

zerodkdj committed:
feat: LottoOutputView클래스추가

- 출력만 담당하는 역할을 LottoOutputView에서 담당
- 사용자 입력을 받는 부분은 LottoConsoleView에서 담당

zerodkdj committed:
refactor: Lotto클래스 등수구하는 메소드 입력값 검증로직추가

seokhongkim reviewed (approved):
동기님, 3단계 미션 피드백 잘 반영해주셨습니다. 💯 
3단계 PR은 이만 머지하도록 하겠습니다.

수정이 조금 더 필요한 부분(숫자 정렬)은 코멘트에 답변으로 남겨 놓았으니 확인 부탁드립니다. 🙏 

4단계 미션에서 뵙겠습니다! 😀
